code/client/scripts/index.js

```diff
@@ -1224,7 +1224,7 @@ function ShareLink(channel, entity, product_imgurl, product_name, product_brand)
                     pin_desc = pin_desc.trim().slice(0, -1)
                 }
             }
-            window.open('https://pinterest.com/pin/create/link/?url=' + encodeURIComponent(url) + '&media=' + encodeURIComponent(product_imgurl) + '&description=' + escape(pin_desc), "_blank");
+            window.open('https://pinterest.com/pin/create/link/?url=' + encodeURIComponent(url.replace("https","http")) + '&media=' + encodeURIComponent(product_imgurl) + '&description=' + escape(pin_desc), "_blank");
```

Comment:
I don't believe this is the problem.  If you click the Pin button, it takes you to the following link:
```
https://www.pinterest.com/pin/create/link/?url=https%253A%2F%2Fbagcupid.com%2Fapp%2Ftags%2F3&media=https%3A%2F%2Fimages-na.ssl-images-amazon.com%2Fimages%2FI%2F51-LfhhdMNL._SL160_.jpg&description=%23style%3A%20clutch
```
If you URL decode everything after the `?url=` bit you get:
```
https%3A//bagcupid.com/app/tags/3&media=https://images-na.ssl-images-amazon.com/images/I/51-LfhhdMNL._SL160_.jpg&description=#style: clutch
```
The problem is that the `:` is double-encoded for some reason, but the rest of the URL isn't double decoded.